Removing unneeded function calls and empty bodies

pi1024e opened the pull request:
One of the things I found strange when looking at the code is seeing how corerun is literally a main function, with main passing ALL of its parameters to that function.

This is confusing; why not main corerun main?!

So, I did just that.

Please let me know if there are any problems.

Thank you

jkotas commented:
build errors

AaronRobinsonMSFT commented on src/jit/compiler.cpp:
```diff
@@ -9750,8 +9749,7 @@ int cLeafIR(Compiler* comp, GenTree* tree)
                             {
                                 case GenTree::GT_REGTAG_REG:
-                                    chars += printf(":%s", comp->compRegVarName(tree->GetRegNum()));
-                                    break;
```
Fall throughs should be avoided. Please add the `break;` back in all cases.

AaronRobinsonMSFT commented on src/jit/compiler.cpp:
```diff
@@ -10692,11 +10685,10 @@ void cNodeIR(Compiler* comp, GenTree* tree)
         chars += cLeafIR(comp, tree);
     }
-    else if (op == GT_LEA)
+    else if !(op == GT_LEA) // Already dumped GT_LEA above.
```
I am not familiar with the `!` outside the parentheses. Is this a new syntax in C++?

pi1024e replied:
I apologize, there were a few commits that should not have slipped in, but did. I fixed this!

pi1024e commented:
A few commits that were not supposed to slipped through, calling the build errors. I have moved them to the correct branch on my fork and removed them from this PR.

jkotas commented:
Thank!
